mypy/fastparse.py

```diff
@@ -84,18 +84,21 @@ def parse(source: Union[str, bytes], fnam: str = None, errors: Errors = None,
 def parse_type_comment(type_comment: str, line: int) -> Type:
     try:
         typ = ast35.parse(type_comment, '<type_comment>', 'eval')
-    except SyntaxError:
-        raise TypeCommentParseError(TYPE_COMMENT_SYNTAX_ERROR, line)
+    except SyntaxError as e:
+        raise TypeCommentParseError(TYPE_COMMENT_SYNTAX_ERROR, line, e.offset)
     else:
         assert isinstance(typ, ast35.Expression)
         return TypeConverter(line=line).visit(typ.body)
 
 
-def with_line(f: Callable[['ASTConverter', T], U]) -> Callable[['ASTConverter', T], U]:
+def with_line_and_column(f: Callable[['ASTConverter', T], U]) -> Callable[['ASTConverter', T], U]:
```

Comment:
done - could totally change the name at some point in the future in a single-purpose PR if it becomes confusing